ehsm_kms_service/function.js

```diff
@@ -34,6 +34,15 @@ const base64_encode = (str) => new Buffer.from(str).toString('base64')
 // base64 decode
 const base64_decode = (base64_str) =>
   new Buffer.from(base64_str, 'base64').toString()
+const is_base64 = (base64_str) => {
+  if (base64_str == undefined ||
+      typeof base64_str != 'string' ||
+      base64_str.length == 0 ||

```

Comment:
const aad = {
  type: 'base64',
  maxLength: MAX_LENGTH,
  required: false,
}

aad='' is valid

Reply:
This is a common function for checking valid base64, not only for aad. 
And "required: false," will cover the case that "aad=''" at line 322 - 328